OpenRA.Game/Activities/Activity.cs

```diff
@@ -99,7 +99,7 @@ protected Activity ChildActivity
 		{
 			get
 			{
-				return childActivity != null && childActivity.State < ActivityState.Done ? childActivity : null;
+				return childActivity != null && childActivity.State != ActivityState.Done ? childActivity : null;
```

Comment:
It would be much better conceptually if we could keep `ActivityState.Canceled` meaning that the activity is also done - how much work do you think it would be to introduce a new `ActivityState.Canceling` and put `Move` in this state until it has completed?

Reply:
Keep in mind: It may not have been originally intended that way, but the current `Canceled` state is in practice already mostly equivalent to the proposed `Canceling` state. So introducing a new `Canceling` state is mostly a renaming exercise. Anyway, I have split the states up now  and used it for `Move` as well, which does help in making `Move` less of an exception.

Reply:
Ah sorry, maybe I should have been more clear. I was thinking that this would then let us keep the ordering of the `ActivityState`s here (`Cancelling` before done, `Cancelled` staying after `Done`) and then the `IsCancelled` check would return true for both `Cancelling` and `Cancelled`.

Reply:
Agreed on the first part; That was what I intended. Fixed.

I don't agree on the second part. That would just be muddling the concepts again. As it is, `IsCanceled` / `IsCanceling` is almost exclusively called from the Tick method of the Activity being cancelled. It just doesn't make sense to me that an activity that is completely done and over should ever get the chance to tick again in the first place. Whenever this clause is triggered, it means the `Activity` is still in the process of cancellation.

Reply:
Hmm, ok, that does make sense. This then also invalidates my earlier argument/request about keeping `Canceling` and `Canceled` as distinct states, so better to have `Canceling` transition to `Done` and do away with the latter.

Reply:
Thinking out loud: is it reasonable to just redefine the meaning of `Canceled` without changing the naming to `Canceling`? I suspect that this reduces us back to exactly what you had here in the first place...

Reply:
I think we could go either way with the name. Considering I assumed this was what `Canceled` meant all along when I first looked into the code, I wouldn't say it is unreasonable. On the other hand, `Canceling` leaves less room for ambiguity.

I wouldn't say we are exactly where we were before. There is still the `Move.Cancel` consolidation I snuck in.